store/src/java/com/zimbra/cs/datasource/imap/RemoteFolder.java

```diff
@@ -71,10 +77,39 @@ public void delete() throws IOException {
 
     public RemoteFolder renameTo(String newName) throws IOException {
         info("renaming folder to '%s'", newName);
-        connection.rename(path, newName);
+
+        // ZBUG-2192
+        // execute only when last folder name's are different
+        // don't change any parent folder names
+        // using "path" as base name we are forming new folder name 
+        // in such a way only last folder get's renamed at a time
+        if (!isSameFolderName(newName)) {
```

Comment:
Avoid negative checks for boolean.

Reply:
@log2akshat Either I add a negative check here or I have to return a negative boolean value from isSameFolderName()
which again is a similar thing. what would be your suggestion to return a negative boolean or continue with the negative check?

Reply:
@log2akshat, Actually all the if-else conditions are required as there are multiple scenarios in which the Folder name can be passed as an input. Hence in order to handle appropriate way of renaming all the if-else conditions are required.
I tried but couldn't find any other approaches to refactor it, if you have any suggestions we can connect and discuss on that

Reply:
@log2akshat for this I have created new method which return the basenewpath and also removed an if condition at L:94 which removed the nested if else block

Reply:
@gabhinavTI suggestion here, make the method `isSameFolderName` other way round as `notSameFolderName` or `folderNameIsChanged` to avoid negative check and from the method return the boolean accordingly.